core/src/main/java/org/apache/iceberg/UpdateRequirement.java

```diff
@@ -199,4 +220,56 @@ public void validate(TableMetadata base) {
       }
     }
   }
+
+  class AssertLastAssignedViewVersionId implements UpdateRequirement {
+    private final int versionId;
+
+    public AssertLastAssignedViewVersionId(int versionId) {
+      this.versionId = versionId;
+    }
+
+    public int versionId() {
+      return versionId;
+    }
+
+    @Override
+    public void validate(TableMetadata base) {
+      // noop
```

Comment:
I think all the places where there are currently noops should throw.

Reply:
updated